packages/compiler-sfc/__tests__/compileScriptRefTransform.spec.ts

```diff
@@ -32,7 +32,17 @@ describe('sfc ref transform', () => {
     // normal declarations left untouched
     expect(content).toMatch(`let c = () => {}`)
     expect(content).toMatch(`let d`)
-    expect(content).toMatch(`return { foo, a, b, c, d, ref, shallowRef }`)
+    expect(content).toMatch(`return (function getReturnedProxy(){`)
+    expect(content).toMatch(`const result = {}`)
+    expect(content).toMatch(`Object.defineProperty(result, 'foo', { get: function() { return foo }, set: function(_val_) { foo = _val_ } })`)
+    expect(content).toMatch(`Object.defineProperty(result, 'a', { get: function() { return a }, set: function(_val_) { a = _val_ } })`)
+    expect(content).toMatch(`Object.defineProperty(result, 'b', { get: function() { return b }, set: function(_val_) { b = _val_ } })`)
+    expect(content).toMatch(`Object.defineProperty(result, 'c', { get: function() { return c }, set: function(_val_) { c = _val_ } })`)
+    expect(content).toMatch(`Object.defineProperty(result, 'd', { get: function() { return d }, set: function(_val_) { d = _val_ } })`)
+    expect(content).toMatch(`Object.defineProperty(result, 'ref', { get: function() { return ref }, set: function(_val_) { ref = _val_ } })`)
+    expect(content).toMatch(`Object.defineProperty(result, 'shallowRef', { get: function() { return shallowRef }, set: function(_val_) { shallowRef = _val_ } })`)
```

Comment:
should not wrap `ref` and `shallowRef` or any other `setup-const` variables.
IMO, just wrap `setup-let` variables will be fine.

Reply:
Yeah actually I considered wrap `setup-let` in the beginning, but I find out there are some benifits -- considering `$setup.var` as reference of `var` in `setup` closure in dev mode, mentally equal with reference in returned render function, if someone modify a `setup-const` var in inline event handler at render func, he can get `Assignment to constant variable.` error from js engine, just like prod mode.

But there might be some affects I didn't notice, will be appreciate to get more suggestion. And just wrap `setup-let` won't be hard, you can decide.

Reply:
Im not sure. lets keep it and waiting for Evan's review.